### PR TITLE
Sort Support for feign pagination for 2.x.x

### PR DESCRIPTION
**Pull request by cbezmen**
PageJacksonModule doesn't support for sort operations. It will serialize and deserialize sort object for spring.

### CODE REVIEW

**Comment by OlgaMaciaszek** [comment]
@cbezmen Please add info to documentation as @ryanjbaxter has suggested [here](https://github.com/spring-cloud/spring-cloud-openfeign/pull/328). It should be added to `spring-cloud-openfeign.adoc`, in AsciiDoc format.

**Comment by spencergibb** [comment]
Ah, this is specific to sorting, not just paging correct?

**Comment by cbezmen** [comment]
Yes, It's just for serialize and deserialize Sort object. Paging (page, size.. etc.) already done by other developer. (https://github.com/spring-cloud/spring-cloud-openfeign/pull/103)

File: spring-cloud-openfeign-core/src/main/java/org/springframework/cloud/openfeign/support/SortJacksonModule.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+import com.fasterxml.jackson.databind.module.SimpleDeserializers;
+import com.fasterxml.jackson.databind.module.SimpleSerializers;
+
+import org.springframework.data.domain.Sort;
+
+/**
+ * This jackson module provides support to add serialize and deserialize for spring
+ * {@link Sort} object.
+ *
+ * @author canbezmen
```
Comment (OlgaMaciaszek) [inline spring-cloud-openfeign-core/src/main/java/org/springframework/cloud/openfeign/support/SortJacksonModule.java:30]: Please use your full name and surname in `@author` clauses in all changed classes.

File: spring-cloud-openfeign-core/src/main/java/org/springframework/cloud/openfeign/support/SortJacksonModule.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+import com.fasterxml.jackson.core.Version;
+import com.fasterxml.jackson.databind.Module;
+import com.fasterxml.jackson.databind.module.SimpleDeserializers;
+import com.fasterxml.jackson.databind.module.SimpleSerializers;
+
+import org.springframework.data.domain.Sort;
+
+/**
+ * This jackson module provides support to add serialize and deserialize for spring
```
Comment (OlgaMaciaszek) [inline spring-cloud-openfeign-core/src/main/java/org/springframework/cloud/openfeign/support/SortJacksonModule.java:27]: Please use correct capitalization: "Jackson", "Spring". Modify this part: "provides support to add serialize and deserialize" to "provides support for serializing and deserializing.

File: spring-cloud-openfeign-core/src/main/java/org/springframework/cloud/openfeign/support/SortJsonComponent.java [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.JsonSerializer;
+import com.fasterxml.jackson.databind.SerializerProvider;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import feign.codec.EncodeException;
+
+import org.springframework.data.domain.Sort;
+
+/**
+ * This class provides support to serialize and deserialize spring {@link Sort} object.
```
Comment (OlgaMaciaszek) [inline spring-cloud-openfeign-core/src/main/java/org/springframework/cloud/openfeign/support/SortJsonComponent.java:37]: Please capitalize "Spring" correctly.

File: spring-cloud-openfeign-core/src/main/java/org/springframework/cloud/openfeign/support/SortJsonComponent.java [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+ *
+ * @author canbezmen
```
Comment (OlgaMaciaszek) [inline spring-cloud-openfeign-core/src/main/java/org/springframework/cloud/openfeign/support/SortJsonComponent.java:39]: Please use your full name and surname in `@authors` clause.

File: spring-cloud-openfeign-core/src/test/java/org/springframework/cloud/openfeign/support/SortJacksonModuleTests.java [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.hasProperty;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.notNullValue;
+
+/**
+ * @author canbezmen
```
Comment (OlgaMaciaszek) [inline spring-cloud-openfeign-core/src/test/java/org/springframework/cloud/openfeign/support/SortJacksonModuleTests.java:41]: Please use your full name and surname in the `@authors` clause.

File: spring-cloud-openfeign-core/src/main/java/org/springframework/cloud/openfeign/support/SortJacksonModule.java (hunk 1 above)
Comment (cbezmen) [reply spring-cloud-openfeign-core/src/main/java/org/springframework/cloud/openfeign/support/SortJacksonModule.java]: done

File: spring-cloud-openfeign-core/src/main/java/org/springframework/cloud/openfeign/support/SortJacksonModule.java (hunk 2 above)
Comment (cbezmen) [reply spring-cloud-openfeign-core/src/main/java/org/springframework/cloud/openfeign/support/SortJacksonModule.java]: done

File: spring-cloud-openfeign-core/src/main/java/org/springframework/cloud/openfeign/support/SortJsonComponent.java (hunk 3 above)
Comment (cbezmen) [reply spring-cloud-openfeign-core/src/main/java/org/springframework/cloud/openfeign/support/SortJsonComponent.java]: done

File: spring-cloud-openfeign-core/src/main/java/org/springframework/cloud/openfeign/support/SortJsonComponent.java (hunk 4 above)
Comment (cbezmen) [reply spring-cloud-openfeign-core/src/main/java/org/springframework/cloud/openfeign/support/SortJsonComponent.java]: done

File: spring-cloud-openfeign-core/src/test/java/org/springframework/cloud/openfeign/support/SortJacksonModuleTests.java (hunk 5 above)
Comment (cbezmen) [reply spring-cloud-openfeign-core/src/test/java/org/springframework/cloud/openfeign/support/SortJacksonModuleTests.java]: done

**Pull request merged**